src/Bicep.Core/TypeSystem/Providers/Az/AzResourceTypeFactory.cs

```diff
@@ -57,7 +57,8 @@ public IEnumerable<FunctionOverload> GetResourceFunctionOverloads(Azure.Bicep.Ty
         }
 
         private TypeSymbol GetTypeSymbol(Azure.Bicep.Types.Concrete.TypeBase serializedType, bool isResourceBodyType, bool isResourceBodyTopLevelPropertyType)
-            => typeCache.GetOrAdd(serializedType, serializedType => ToTypeSymbol(serializedType, isResourceBodyType, isResourceBodyTopLevelPropertyType));
+            => typeCache.GetOrAdd((serializedType, isResourceBodyType, isResourceBodyTopLevelPropertyType),
```

Comment:
Could you add a comment on the function to note that the arguments should match the cache keys? To try and reduce the chance of the same issue cropping up again in the future.